[lldb] Extend information for failed connection for gdb server

ita-sc opened the pull request:
Before:
```
(lldb) r
error: connect remote failed (Failed to connect port)
error: Failed to connect port
```

After the patch:
```
(lldb) r
error: connect remote failed (Failed to connect localhost:47140)
error: Failed to connect localhost:47140
```

JDevlieghere commented on lldb/source/Host/common/TCPSocket.cpp:
```diff
@@ -169,7 +169,8 @@ Status TCPSocket::Connect(llvm::StringRef name) {
   }
 
-  error = Status::FromErrorString("Failed to connect port");
+  error = Status::FromErrorStringWithFormatv(
+      "Failed to connect {0}:{1}", host_port->hostname, host_port->port);
```
```suggestion
  error = Status::FromErrorStringWithFormatv(
      "Failed to connect to {0}:{1}", host_port->hostname, host_port->port);
```

ita-sc replied:
Yep, fixed

JDevlieghere reviewed (approved):
LGTM modulo (pre-existing) typo.

ita-sc committed:
Extend information for failed connection for gdb server

Before:
```
(lldb) r
error: connect remote failed (Failed to connect port)
error: Failed to connect port
```

After the patch:
```
(lldb) r
error: connect remote failed (Failed to connect to localhost:47140)
error: Failed to connect to localhost:47140
```
